Python/syntax_test_python.py

```diff
@@ -1613,6 +1614,21 @@ class MyClass(Inherited, \
     pass
 
 
+class DataClass(TypedDict, None, total=False, True=False):
+#     ^^^^^^^^^ entity.name.class.python
+#               ^^^^^^^^^ entity.other.inherited-class.python
+#                        ^ punctuation.separator.inheritance.python
+#                          ^^^^ constant.language.null.python
+#                              ^ punctuation.separator.inheritance.python
+#                                ^^^^^ variable.parameter.class-inheritance.python
+#                                     ^ keyword.operator.assignment.python
+#                                      ^^^^^ constant.language.boolean.python
+#                                           ^ punctuation.separator.inheritance.python
+#                                             ^^^^ constant.language.boolean.python
```

Comment:
> ... is meant to test that the identifier used here is reserved? ...

Eaxactly.

> introducing an `invalid_targets` variable

Would be an option as language constants are obviously not to be used as assignment targets. Just a question of how much effort to put into all the illegal scoping as it can easily become quite complex in general. pyright clearly marks it illegal, already.